apps/extension/src/ui/atoms/chaindata.ts

```diff
@@ -322,7 +329,31 @@ export const allTokensMapState = atom<TokenList>({
   effects: [
     ({ setSelf }) => {
       log.debug("allTokensMapState.init")
-      const sub = chaindataProvider.tokensListObservable.subscribe(setSelf)
+      const sub = chaindataProvider.tokensListObservable.subscribe((tokensList) => {
+        // ignore native tokens duplicates, this will also ignore duplicate balances
+        // this is necessary for users that had created custom native tokens prior we changed native token ids (v1.21.0)
+        // note: this cleaup takes 3ms on my machine
```

Comment:
I think regardless of performance impact, this would be better off in a migration, since it should only have to happen once, and then the code can live in a nice easily maintainable place